src/main/java/io/nats/client/impl/NatsMessageConsumer.java

```diff
@@ -19,9 +19,12 @@
 import java.io.IOException;
 
 class NatsMessageConsumer extends NatsMessageConsumerBase implements PullManagerObserver {
-    protected final PullRequestOptions rePullPro;
+    protected final ConsumeOptions opts;
```

Comment:
making the PullRequestOptions on the fly, so need to keep the original user options around.